src/HealthGPS.Console/model_parser.cpp

```diff
@@ -141,59 +141,107 @@ std::unique_ptr<hgps::StaticLinearModelDefinition>
+
+        // Check intervention policy covariance matrix column name matches risk factor name.
+        auto policy_column_name = policy_covariance_table.column(i).name();
+        if (!hgps::core::case_insensitive::equals(key, policy_column_name)) {
+            throw hgps::core::HgpsException{
+                fmt::format("Risk factor {} name ({}) does not match intervention "
+                            "policy covariance matrix column {} name ({})",
+                            i, key, i, policy_column_name)};
+        }
+
+        // Write intervention policy data structures.
+        policy_models.emplace_back(std::move(policy_model));
+        policy_ranges.emplace_back(policy_json_params["Range"].get<hgps::core::DoubleInterval>());
+        for (size_t j = 0; j < policy_covariance_table.num_rows(); j++) {
+            policy_covariance(i, j) =
+                std::any_cast<double>(policy_covariance_table.column(i).value(j));
         }
 
         // Increment table column index.
         i++;
     }
 
-    // Check correlation matrix column count matches risk factor count.
-    if (opt["RiskFactorModels"].size() != correlations_table.num_columns()) {
+    // Check risk factor correlation matrix column count matches risk factor count.
+    if (opt["RiskFactorModels"].size() != correlation_table.num_columns()) {
+        throw hgps::core::HgpsException{fmt::format("Risk factor count ({}) does not match risk "
+                                                    "factor correlation matrix column count ({})",
+                                                    opt["RiskFactorModels"].size(),
+                                                    correlation_table.num_columns())};
+    }
+
+    // Compute Cholesky decomposition of the risk factor correlation matrix.
+    auto cholesky = Eigen::MatrixXd{Eigen::LLT<Eigen::MatrixXd>{correlation}.matrixL()};
+
+    // Check intervention policy covariance matrix column count matches risk factor count.
+    if (opt["RiskFactorModels"].size() != policy_covariance_table.num_columns()) {
         throw hgps::core::HgpsException{
-            fmt::format("Risk factor count ({}) does not match correlation "
-                        "matrix column count ({})",
-                        opt["RiskFactorModels"].size(), correlations_table.num_columns())};
+            fmt::format("Risk factor count ({}) does not match intervention "
+                        "policy covariance matrix column count ({})",
+                        opt["RiskFactorModels"].size(), policy_covariance_table.num_columns())};
     }
 
-    // Compute Cholesky decomposition of correlation matrix.
-    auto cholesky = Eigen::MatrixXd{Eigen::LLT<Eigen::MatrixXd>{correlations}.matrixL()};
+    // Compute Cholesky decomposition of the intervention policy covariance matrix.
+    auto policy_cholesky =
+        Eigen::MatrixXd{Eigen::LLT<Eigen::MatrixXd>{policy_covariance}.matrixL()};
```

Comment:
Are you using `policy_cholesky` or `cholesky` in this function? If not, you could skip casting the type here and instead of ``std::move`'ing it out of the function, just let it be copied, which will achieve the same thing.

Reply:
i'm getting `std::bad_alloc` if I try to do this. I will leave it be for now.

Reply:
Ah ok. Nvm then